{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All Samples are here:\n",
    "all_matched_samples.gct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "samples_file = \"all_matched_samples.gct\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(55490, 224)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "df = pd.read_table(samples_file, skiprows=2, index_col=0)\n",
    "Description = df['Description']\n",
    "df.drop('Description', axis=1, inplace=True)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"well well-sm\">\n",
    "Applying a varaiance filter. Filtering out genes with lower standar deviation than 100. Mean std is about 874 for this dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "39151 genes were filtered out -- they had a Standard Deviation lower than 100 (mean std was ~874)\n",
      "New gene lengths is 16339\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>TCGA-A7-A0CE-01</th>\n",
       "      <th>TCGA-A7-A0CE-11</th>\n",
       "      <th>TCGA-A7-A0CH-01</th>\n",
       "      <th>TCGA-A7-A0CH-11</th>\n",
       "      <th>TCGA-A7-A0D9-01</th>\n",
       "      <th>TCGA-A7-A0D9-11</th>\n",
       "      <th>TCGA-A7-A0DB-01</th>\n",
       "      <th>TCGA-A7-A0DB-11</th>\n",
       "      <th>TCGA-A7-A0DC-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-E9-A1RF-01</th>\n",
       "      <th>TCGA-E9-A1RF-11</th>\n",
       "      <th>TCGA-E9-A1RH-01</th>\n",
       "      <th>TCGA-E9-A1RH-11</th>\n",
       "      <th>TCGA-E9-A1RI-01</th>\n",
       "      <th>TCGA-E9-A1RI-11</th>\n",
       "      <th>TCGA-GI-A2C8-01</th>\n",
       "      <th>TCGA-GI-A2C8-11</th>\n",
       "      <th>TCGA-GI-A2C9-01</th>\n",
       "      <th>TCGA-GI-A2C9-11</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>tetraspanin 6</td>\n",
       "      <td>8462.0</td>\n",
       "      <td>5404.0</td>\n",
       "      <td>3036.0</td>\n",
       "      <td>5030.0</td>\n",
       "      <td>2679.0</td>\n",
       "      <td>3616.0</td>\n",
       "      <td>2203.0</td>\n",
       "      <td>2425.0</td>\n",
       "      <td>4094.0</td>\n",
       "      <td>...</td>\n",
       "      <td>676.0</td>\n",
       "      <td>4544.0</td>\n",
       "      <td>327.0</td>\n",
       "      <td>3581.0</td>\n",
       "      <td>1980.0</td>\n",
       "      <td>7639.0</td>\n",
       "      <td>8586.0</td>\n",
       "      <td>422.0</td>\n",
       "      <td>3944.0</td>\n",
       "      <td>4849.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNMD</th>\n",
       "      <td>tenomodulin</td>\n",
       "      <td>54.0</td>\n",
       "      <td>320.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2116.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>3616.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>304.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.0</td>\n",
       "      <td>1881.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>770.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3541.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPM1</th>\n",
       "      <td>dolichyl-phosphate mannosyltransferase subunit...</td>\n",
       "      <td>2408.0</td>\n",
       "      <td>2472.0</td>\n",
       "      <td>1497.0</td>\n",
       "      <td>1611.0</td>\n",
       "      <td>4135.0</td>\n",
       "      <td>1254.0</td>\n",
       "      <td>1131.0</td>\n",
       "      <td>1137.0</td>\n",
       "      <td>3050.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3732.0</td>\n",
       "      <td>1565.0</td>\n",
       "      <td>6267.0</td>\n",
       "      <td>1615.0</td>\n",
       "      <td>2885.0</td>\n",
       "      <td>1999.0</td>\n",
       "      <td>2687.0</td>\n",
       "      <td>1675.0</td>\n",
       "      <td>2708.0</td>\n",
       "      <td>2290.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCYL3</th>\n",
       "      <td>SCY1 like pseudokinase 3</td>\n",
       "      <td>1518.0</td>\n",
       "      <td>1483.0</td>\n",
       "      <td>3030.0</td>\n",
       "      <td>1154.0</td>\n",
       "      <td>1852.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>1919.0</td>\n",
       "      <td>687.0</td>\n",
       "      <td>1906.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3155.0</td>\n",
       "      <td>1356.0</td>\n",
       "      <td>775.0</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>2538.0</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>3240.0</td>\n",
       "      <td>751.0</td>\n",
       "      <td>1524.0</td>\n",
       "      <td>1265.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C1orf112</th>\n",
       "      <td>chromosome 1 open reading frame 112</td>\n",
       "      <td>2056.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>730.0</td>\n",
       "      <td>252.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>725.0</td>\n",
       "      <td>241.0</td>\n",
       "      <td>846.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1010.0</td>\n",
       "      <td>294.0</td>\n",
       "      <td>770.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>741.0</td>\n",
       "      <td>290.0</td>\n",
       "      <td>617.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>657.0</td>\n",
       "      <td>343.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGR</th>\n",
       "      <td>FGR proto-oncogene, Src family tyrosine kinase</td>\n",
       "      <td>287.0</td>\n",
       "      <td>869.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>1536.0</td>\n",
       "      <td>201.0</td>\n",
       "      <td>969.0</td>\n",
       "      <td>846.0</td>\n",
       "      <td>1517.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>...</td>\n",
       "      <td>910.0</td>\n",
       "      <td>1006.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>578.0</td>\n",
       "      <td>399.0</td>\n",
       "      <td>377.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>422.0</td>\n",
       "      <td>969.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFH</th>\n",
       "      <td>complement factor H</td>\n",
       "      <td>1663.0</td>\n",
       "      <td>5760.0</td>\n",
       "      <td>1531.0</td>\n",
       "      <td>4695.0</td>\n",
       "      <td>1672.0</td>\n",
       "      <td>14103.0</td>\n",
       "      <td>2949.0</td>\n",
       "      <td>8647.0</td>\n",
       "      <td>772.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5382.0</td>\n",
       "      <td>24121.0</td>\n",
       "      <td>2915.0</td>\n",
       "      <td>13543.0</td>\n",
       "      <td>4207.0</td>\n",
       "      <td>19118.0</td>\n",
       "      <td>4565.0</td>\n",
       "      <td>3231.0</td>\n",
       "      <td>2747.0</td>\n",
       "      <td>22260.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUCA2</th>\n",
       "      <td>fucosidase, alpha-L- 2, plasma</td>\n",
       "      <td>10475.0</td>\n",
       "      <td>4397.0</td>\n",
       "      <td>1772.0</td>\n",
       "      <td>3930.0</td>\n",
       "      <td>3733.0</td>\n",
       "      <td>3688.0</td>\n",
       "      <td>1737.0</td>\n",
       "      <td>3963.0</td>\n",
       "      <td>2124.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3559.0</td>\n",
       "      <td>3695.0</td>\n",
       "      <td>2814.0</td>\n",
       "      <td>3660.0</td>\n",
       "      <td>2639.0</td>\n",
       "      <td>3746.0</td>\n",
       "      <td>5589.0</td>\n",
       "      <td>726.0</td>\n",
       "      <td>4024.0</td>\n",
       "      <td>3422.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCLC</th>\n",
       "      <td>glutamate-cysteine ligase catalytic subunit</td>\n",
       "      <td>2330.0</td>\n",
       "      <td>2775.0</td>\n",
       "      <td>3522.0</td>\n",
       "      <td>3217.0</td>\n",
       "      <td>2527.0</td>\n",
       "      <td>4155.0</td>\n",
       "      <td>2814.0</td>\n",
       "      <td>2833.0</td>\n",
       "      <td>9453.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2968.0</td>\n",
       "      <td>5097.0</td>\n",
       "      <td>4311.0</td>\n",
       "      <td>3845.0</td>\n",
       "      <td>3539.0</td>\n",
       "      <td>7854.0</td>\n",
       "      <td>3397.0</td>\n",
       "      <td>1244.0</td>\n",
       "      <td>2673.0</td>\n",
       "      <td>2536.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFYA</th>\n",
       "      <td>nuclear transcription factor Y subunit alpha</td>\n",
       "      <td>2776.0</td>\n",
       "      <td>2060.0</td>\n",
       "      <td>3629.0</td>\n",
       "      <td>1645.0</td>\n",
       "      <td>3048.0</td>\n",
       "      <td>1181.0</td>\n",
       "      <td>2121.0</td>\n",
       "      <td>1175.0</td>\n",
       "      <td>3308.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3704.0</td>\n",
       "      <td>2025.0</td>\n",
       "      <td>5660.0</td>\n",
       "      <td>1556.0</td>\n",
       "      <td>3410.0</td>\n",
       "      <td>1776.0</td>\n",
       "      <td>1798.0</td>\n",
       "      <td>938.0</td>\n",
       "      <td>4720.0</td>\n",
       "      <td>3317.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STPG1</th>\n",
       "      <td>sperm tail PG-rich repeat containing 1</td>\n",
       "      <td>1162.0</td>\n",
       "      <td>845.0</td>\n",
       "      <td>904.0</td>\n",
       "      <td>855.0</td>\n",
       "      <td>438.0</td>\n",
       "      <td>666.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>277.0</td>\n",
       "      <td>...</td>\n",
       "      <td>704.0</td>\n",
       "      <td>591.0</td>\n",
       "      <td>328.0</td>\n",
       "      <td>590.0</td>\n",
       "      <td>913.0</td>\n",
       "      <td>723.0</td>\n",
       "      <td>556.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1418.0</td>\n",
       "      <td>922.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NIPAL3</th>\n",
       "      <td>NIPA like domain containing 3</td>\n",
       "      <td>5432.0</td>\n",
       "      <td>4848.0</td>\n",
       "      <td>3319.0</td>\n",
       "      <td>2896.0</td>\n",
       "      <td>2720.0</td>\n",
       "      <td>1321.0</td>\n",
       "      <td>1927.0</td>\n",
       "      <td>1200.0</td>\n",
       "      <td>4931.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2479.0</td>\n",
       "      <td>1388.0</td>\n",
       "      <td>1356.0</td>\n",
       "      <td>2201.0</td>\n",
       "      <td>3485.0</td>\n",
       "      <td>1850.0</td>\n",
       "      <td>2534.0</td>\n",
       "      <td>889.0</td>\n",
       "      <td>7065.0</td>\n",
       "      <td>5090.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LAS1L</th>\n",
       "      <td>LAS1 like, ribosome biogenesis factor</td>\n",
       "      <td>4739.0</td>\n",
       "      <td>3778.0</td>\n",
       "      <td>2394.0</td>\n",
       "      <td>2944.0</td>\n",
       "      <td>3330.0</td>\n",
       "      <td>1848.0</td>\n",
       "      <td>967.0</td>\n",
       "      <td>1470.0</td>\n",
       "      <td>2509.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5322.0</td>\n",
       "      <td>1665.0</td>\n",
       "      <td>8071.0</td>\n",
       "      <td>2031.0</td>\n",
       "      <td>3179.0</td>\n",
       "      <td>1638.0</td>\n",
       "      <td>2738.0</td>\n",
       "      <td>593.0</td>\n",
       "      <td>3153.0</td>\n",
       "      <td>2688.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENPP4</th>\n",
       "      <td>ectonucleotide pyrophosphatase/phosphodiestera...</td>\n",
       "      <td>946.0</td>\n",
       "      <td>1168.0</td>\n",
       "      <td>2695.0</td>\n",
       "      <td>1759.0</td>\n",
       "      <td>2235.0</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>1715.0</td>\n",
       "      <td>3074.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>3793.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>2605.0</td>\n",
       "      <td>1450.0</td>\n",
       "      <td>4956.0</td>\n",
       "      <td>6140.0</td>\n",
       "      <td>3424.0</td>\n",
       "      <td>1753.0</td>\n",
       "      <td>2302.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEMA3F</th>\n",
       "      <td>semaphorin 3F</td>\n",
       "      <td>3486.0</td>\n",
       "      <td>5125.0</td>\n",
       "      <td>3911.0</td>\n",
       "      <td>3239.0</td>\n",
       "      <td>10121.0</td>\n",
       "      <td>1133.0</td>\n",
       "      <td>2968.0</td>\n",
       "      <td>1729.0</td>\n",
       "      <td>3401.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7643.0</td>\n",
       "      <td>1205.0</td>\n",
       "      <td>2847.0</td>\n",
       "      <td>2061.0</td>\n",
       "      <td>5777.0</td>\n",
       "      <td>1941.0</td>\n",
       "      <td>6497.0</td>\n",
       "      <td>553.0</td>\n",
       "      <td>1331.0</td>\n",
       "      <td>3759.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFTR</th>\n",
       "      <td>cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>279.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANKIB1</th>\n",
       "      <td>ankyrin repeat and IBR domain containing 1</td>\n",
       "      <td>7688.0</td>\n",
       "      <td>3182.0</td>\n",
       "      <td>3500.0</td>\n",
       "      <td>2429.0</td>\n",
       "      <td>7237.0</td>\n",
       "      <td>3496.0</td>\n",
       "      <td>5265.0</td>\n",
       "      <td>3122.0</td>\n",
       "      <td>9206.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3608.0</td>\n",
       "      <td>5260.0</td>\n",
       "      <td>10522.0</td>\n",
       "      <td>4264.0</td>\n",
       "      <td>8070.0</td>\n",
       "      <td>6739.0</td>\n",
       "      <td>6016.0</td>\n",
       "      <td>2298.0</td>\n",
       "      <td>6059.0</td>\n",
       "      <td>4562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CYP51A1</th>\n",
       "      <td>cytochrome P450 family 51 subfamily A member 1</td>\n",
       "      <td>336.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>439.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>572.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>882.0</td>\n",
       "      <td>...</td>\n",
       "      <td>149.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRIT1</th>\n",
       "      <td>KRIT1, ankyrin repeat containing</td>\n",
       "      <td>3430.0</td>\n",
       "      <td>2505.0</td>\n",
       "      <td>1893.0</td>\n",
       "      <td>1484.0</td>\n",
       "      <td>4495.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>3068.0</td>\n",
       "      <td>1619.0</td>\n",
       "      <td>3320.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1403.0</td>\n",
       "      <td>2650.0</td>\n",
       "      <td>2409.0</td>\n",
       "      <td>2316.0</td>\n",
       "      <td>3391.0</td>\n",
       "      <td>2540.0</td>\n",
       "      <td>2427.0</td>\n",
       "      <td>1729.0</td>\n",
       "      <td>2351.0</td>\n",
       "      <td>2927.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAD52</th>\n",
       "      <td>RAD52 homolog, DNA repair protein</td>\n",
       "      <td>930.0</td>\n",
       "      <td>573.0</td>\n",
       "      <td>389.0</td>\n",
       "      <td>375.0</td>\n",
       "      <td>654.0</td>\n",
       "      <td>380.0</td>\n",
       "      <td>393.0</td>\n",
       "      <td>271.0</td>\n",
       "      <td>742.0</td>\n",
       "      <td>...</td>\n",
       "      <td>562.0</td>\n",
       "      <td>590.0</td>\n",
       "      <td>365.0</td>\n",
       "      <td>429.0</td>\n",
       "      <td>657.0</td>\n",
       "      <td>467.0</td>\n",
       "      <td>439.0</td>\n",
       "      <td>498.0</td>\n",
       "      <td>872.0</td>\n",
       "      <td>843.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BAD</th>\n",
       "      <td>BCL2 associated agonist of cell death</td>\n",
       "      <td>863.0</td>\n",
       "      <td>2586.0</td>\n",
       "      <td>1004.0</td>\n",
       "      <td>2604.0</td>\n",
       "      <td>1768.0</td>\n",
       "      <td>1378.0</td>\n",
       "      <td>700.0</td>\n",
       "      <td>1099.0</td>\n",
       "      <td>374.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1738.0</td>\n",
       "      <td>807.0</td>\n",
       "      <td>835.0</td>\n",
       "      <td>1360.0</td>\n",
       "      <td>1526.0</td>\n",
       "      <td>768.0</td>\n",
       "      <td>1500.0</td>\n",
       "      <td>642.0</td>\n",
       "      <td>2554.0</td>\n",
       "      <td>1731.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LAP3</th>\n",
       "      <td>leucine aminopeptidase 3</td>\n",
       "      <td>7414.0</td>\n",
       "      <td>9073.0</td>\n",
       "      <td>6335.0</td>\n",
       "      <td>6968.0</td>\n",
       "      <td>4508.0</td>\n",
       "      <td>4866.0</td>\n",
       "      <td>3474.0</td>\n",
       "      <td>6087.0</td>\n",
       "      <td>5420.0</td>\n",
       "      <td>...</td>\n",
       "      <td>11987.0</td>\n",
       "      <td>5799.0</td>\n",
       "      <td>6567.0</td>\n",
       "      <td>6457.0</td>\n",
       "      <td>8764.0</td>\n",
       "      <td>6895.0</td>\n",
       "      <td>4841.0</td>\n",
       "      <td>4281.0</td>\n",
       "      <td>6757.0</td>\n",
       "      <td>7810.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD99</th>\n",
       "      <td>CD99 molecule (Xg blood group)</td>\n",
       "      <td>5696.0</td>\n",
       "      <td>12627.0</td>\n",
       "      <td>7232.0</td>\n",
       "      <td>20444.0</td>\n",
       "      <td>7911.0</td>\n",
       "      <td>18066.0</td>\n",
       "      <td>3482.0</td>\n",
       "      <td>22271.0</td>\n",
       "      <td>1572.0</td>\n",
       "      <td>...</td>\n",
       "      <td>15759.0</td>\n",
       "      <td>18304.0</td>\n",
       "      <td>11117.0</td>\n",
       "      <td>16453.0</td>\n",
       "      <td>13723.0</td>\n",
       "      <td>17053.0</td>\n",
       "      <td>11737.0</td>\n",
       "      <td>5089.0</td>\n",
       "      <td>8724.0</td>\n",
       "      <td>12885.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HS3ST1</th>\n",
       "      <td>heparan sulfate-glucosamine 3-sulfotransferase 1</td>\n",
       "      <td>126.0</td>\n",
       "      <td>347.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>369.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>189.0</td>\n",
       "      <td>234.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>...</td>\n",
       "      <td>181.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>240.0</td>\n",
       "      <td>264.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>252.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>1301.0</td>\n",
       "      <td>510.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AOC1</th>\n",
       "      <td>amine oxidase, copper containing 1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1097.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>64.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAD1L1</th>\n",
       "      <td>MAD1 mitotic arrest deficient like 1</td>\n",
       "      <td>3323.0</td>\n",
       "      <td>1909.0</td>\n",
       "      <td>936.0</td>\n",
       "      <td>1696.0</td>\n",
       "      <td>1660.0</td>\n",
       "      <td>979.0</td>\n",
       "      <td>740.0</td>\n",
       "      <td>940.0</td>\n",
       "      <td>1015.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1164.0</td>\n",
       "      <td>456.0</td>\n",
       "      <td>1452.0</td>\n",
       "      <td>860.0</td>\n",
       "      <td>851.0</td>\n",
       "      <td>523.0</td>\n",
       "      <td>1314.0</td>\n",
       "      <td>805.0</td>\n",
       "      <td>1432.0</td>\n",
       "      <td>1631.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LASP1</th>\n",
       "      <td>LIM and SH3 protein 1</td>\n",
       "      <td>13231.0</td>\n",
       "      <td>18957.0</td>\n",
       "      <td>17489.0</td>\n",
       "      <td>23386.0</td>\n",
       "      <td>13768.0</td>\n",
       "      <td>19853.0</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>16959.0</td>\n",
       "      <td>15047.0</td>\n",
       "      <td>...</td>\n",
       "      <td>26291.0</td>\n",
       "      <td>13435.0</td>\n",
       "      <td>31483.0</td>\n",
       "      <td>14566.0</td>\n",
       "      <td>32197.0</td>\n",
       "      <td>18844.0</td>\n",
       "      <td>13011.0</td>\n",
       "      <td>3013.0</td>\n",
       "      <td>10706.0</td>\n",
       "      <td>16794.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNX11</th>\n",
       "      <td>sorting nexin 11</td>\n",
       "      <td>1362.0</td>\n",
       "      <td>1942.0</td>\n",
       "      <td>1290.0</td>\n",
       "      <td>1822.0</td>\n",
       "      <td>1482.0</td>\n",
       "      <td>1613.0</td>\n",
       "      <td>640.0</td>\n",
       "      <td>1642.0</td>\n",
       "      <td>1240.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3997.0</td>\n",
       "      <td>2195.0</td>\n",
       "      <td>1580.0</td>\n",
       "      <td>1997.0</td>\n",
       "      <td>2927.0</td>\n",
       "      <td>2145.0</td>\n",
       "      <td>1949.0</td>\n",
       "      <td>755.0</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>1447.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM176A</th>\n",
       "      <td>transmembrane protein 176A</td>\n",
       "      <td>2351.0</td>\n",
       "      <td>5231.0</td>\n",
       "      <td>234.0</td>\n",
       "      <td>2252.0</td>\n",
       "      <td>2446.0</td>\n",
       "      <td>1352.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5008.0</td>\n",
       "      <td>1659.0</td>\n",
       "      <td>487.0</td>\n",
       "      <td>2157.0</td>\n",
       "      <td>1641.0</td>\n",
       "      <td>2614.0</td>\n",
       "      <td>13815.0</td>\n",
       "      <td>572.0</td>\n",
       "      <td>939.0</td>\n",
       "      <td>3105.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M6PR</th>\n",
       "      <td>mannose-6-phosphate receptor, cation dependent</td>\n",
       "      <td>9044.0</td>\n",
       "      <td>4161.0</td>\n",
       "      <td>4002.0</td>\n",
       "      <td>3960.0</td>\n",
       "      <td>4002.0</td>\n",
       "      <td>3210.0</td>\n",
       "      <td>2841.0</td>\n",
       "      <td>3344.0</td>\n",
       "      <td>6640.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5028.0</td>\n",
       "      <td>4226.0</td>\n",
       "      <td>4758.0</td>\n",
       "      <td>3436.0</td>\n",
       "      <td>5041.0</td>\n",
       "      <td>3540.0</td>\n",
       "      <td>4361.0</td>\n",
       "      <td>1764.0</td>\n",
       "      <td>12882.0</td>\n",
       "      <td>5210.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC133065.6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>77.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>147.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>...</td>\n",
       "      <td>160.0</td>\n",
       "      <td>296.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>185.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>356.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC022107.1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>363.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>115.0</td>\n",
       "      <td>501.0</td>\n",
       "      <td>...</td>\n",
       "      <td>158.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>496.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>439.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>292.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>148.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC005831.1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>429.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>1013.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>260.0</td>\n",
       "      <td>...</td>\n",
       "      <td>465.0</td>\n",
       "      <td>152.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>534.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC026401.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>461.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>307.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>...</td>\n",
       "      <td>342.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>798.0</td>\n",
       "      <td>437.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC093908.2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>518.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>240.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>...</td>\n",
       "      <td>169.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>2353.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>346.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>102.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR4697HG</th>\n",
       "      <td>MIR4697 host gene</td>\n",
       "      <td>28.0</td>\n",
       "      <td>970.0</td>\n",
       "      <td>228.0</td>\n",
       "      <td>390.0</td>\n",
       "      <td>1141.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>259.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>889.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>680.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC233723.2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>107.0</td>\n",
       "      <td>387.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>325.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>...</td>\n",
       "      <td>179.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>340.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>254.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AP001528.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>46.0</td>\n",
       "      <td>576.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>2293.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>3013.0</td>\n",
       "      <td>343.0</td>\n",
       "      <td>2212.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>...</td>\n",
       "      <td>85.0</td>\n",
       "      <td>5370.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>3617.0</td>\n",
       "      <td>283.0</td>\n",
       "      <td>7830.0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>368.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>827.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC000123.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>289.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>678.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>346.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>775.0</td>\n",
       "      <td>...</td>\n",
       "      <td>202.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>263.0</td>\n",
       "      <td>323.0</td>\n",
       "      <td>453.0</td>\n",
       "      <td>614.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>316.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Z95331.1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>314.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AP000648.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>402.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>619.0</td>\n",
       "      <td>...</td>\n",
       "      <td>262.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>237.0</td>\n",
       "      <td>201.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>344.0</td>\n",
       "      <td>331.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC245369.3</th>\n",
       "      <td>Immunoglobulin heavy variable 1-69D</td>\n",
       "      <td>324.0</td>\n",
       "      <td>658.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1898.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1386.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1385.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>708.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC163</th>\n",
       "      <td>coiled-coil domain containing 163</td>\n",
       "      <td>299.0</td>\n",
       "      <td>483.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>...</td>\n",
       "      <td>251.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>483.0</td>\n",
       "      <td>217.0</td>\n",
       "      <td>339.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>388.0</td>\n",
       "      <td>210.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH3PXD2A-AS1</th>\n",
       "      <td>SH3PXD2A antisense RNA 1</td>\n",
       "      <td>13.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LINC01232</th>\n",
       "      <td>long intergenic non-protein coding RNA 1232</td>\n",
       "      <td>144.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>209.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>...</td>\n",
       "      <td>149.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>1060.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>215.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>277.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGR1</th>\n",
       "      <td>PAXIP1 associated glutamate rich protein 1</td>\n",
       "      <td>669.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>419.0</td>\n",
       "      <td>382.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>773.0</td>\n",
       "      <td>...</td>\n",
       "      <td>262.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>449.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>302.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>455.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>496.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LINC00294</th>\n",
       "      <td>long intergenic non-protein coding RNA 294</td>\n",
       "      <td>783.0</td>\n",
       "      <td>724.0</td>\n",
       "      <td>303.0</td>\n",
       "      <td>818.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>847.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>806.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>...</td>\n",
       "      <td>781.0</td>\n",
       "      <td>1423.0</td>\n",
       "      <td>772.0</td>\n",
       "      <td>943.0</td>\n",
       "      <td>791.0</td>\n",
       "      <td>1121.0</td>\n",
       "      <td>3464.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>1420.0</td>\n",
       "      <td>950.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSEC</th>\n",
       "      <td>G-quadruplex forming sequence containing lncRNA</td>\n",
       "      <td>353.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>555.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>...</td>\n",
       "      <td>98.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>573.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1074.0</td>\n",
       "      <td>128.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXCUT</th>\n",
       "      <td>FOXC1 upstream transcript (non-protein coding)</td>\n",
       "      <td>195.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N4BP2L2-IT2</th>\n",
       "      <td>N4BPL2 intronic transcript 2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>115.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>369.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>...</td>\n",
       "      <td>277.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>149.0</td>\n",
       "      <td>187.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRG-AS1</th>\n",
       "      <td>T-cell receptor gamma locus antisense RNA 1</td>\n",
       "      <td>23.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>...</td>\n",
       "      <td>128.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>106.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC007878.1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>32.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>355.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>514.0</td>\n",
       "      <td>...</td>\n",
       "      <td>59.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>162.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AL133410.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>81.0</td>\n",
       "      <td>248.0</td>\n",
       "      <td>316.0</td>\n",
       "      <td>283.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>...</td>\n",
       "      <td>107.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>270.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>299.0</td>\n",
       "      <td>1301.0</td>\n",
       "      <td>337.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>303.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC018645.3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>174.0</td>\n",
       "      <td>298.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>1168.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>...</td>\n",
       "      <td>144.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>148.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HELLPAR</th>\n",
       "      <td>HELLP associated long non-coding RNA</td>\n",
       "      <td>75.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>681.0</td>\n",
       "      <td>...</td>\n",
       "      <td>79.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNHG4</th>\n",
       "      <td>small nucleolar RNA host gene 4</td>\n",
       "      <td>126.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>...</td>\n",
       "      <td>237.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>222.0</td>\n",
       "      <td>226.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CICP14</th>\n",
       "      <td>capicua transcriptional repressor pseudogene 14</td>\n",
       "      <td>518.0</td>\n",
       "      <td>452.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>380.0</td>\n",
       "      <td>617.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>357.0</td>\n",
       "      <td>358.0</td>\n",
       "      <td>...</td>\n",
       "      <td>329.0</td>\n",
       "      <td>268.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>224.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>410.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>260.0</td>\n",
       "      <td>691.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EBLN3P</th>\n",
       "      <td>endogenous Bornavirus-like nucleoprotein 3, ps...</td>\n",
       "      <td>3621.0</td>\n",
       "      <td>3397.0</td>\n",
       "      <td>3963.0</td>\n",
       "      <td>2696.0</td>\n",
       "      <td>4726.0</td>\n",
       "      <td>2465.0</td>\n",
       "      <td>4153.0</td>\n",
       "      <td>2378.0</td>\n",
       "      <td>8902.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3758.0</td>\n",
       "      <td>3907.0</td>\n",
       "      <td>3538.0</td>\n",
       "      <td>3913.0</td>\n",
       "      <td>9734.0</td>\n",
       "      <td>4670.0</td>\n",
       "      <td>5983.0</td>\n",
       "      <td>1940.0</td>\n",
       "      <td>4337.0</td>\n",
       "      <td>4834.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AP000851.2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1555.0</td>\n",
       "      <td>130.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LINC01230</th>\n",
       "      <td>long intergenic non-protein coding RNA 1230</td>\n",
       "      <td>5.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>613.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>329.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16339 rows × 225 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Description  \\\n",
       "Name                                                              \n",
       "TSPAN6                                            tetraspanin 6   \n",
       "TNMD                                                tenomodulin   \n",
       "DPM1          dolichyl-phosphate mannosyltransferase subunit...   \n",
       "SCYL3                                  SCY1 like pseudokinase 3   \n",
       "C1orf112                    chromosome 1 open reading frame 112   \n",
       "FGR              FGR proto-oncogene, Src family tyrosine kinase   \n",
       "CFH                                         complement factor H   \n",
       "FUCA2                            fucosidase, alpha-L- 2, plasma   \n",
       "GCLC                glutamate-cysteine ligase catalytic subunit   \n",
       "NFYA               nuclear transcription factor Y subunit alpha   \n",
       "STPG1                    sperm tail PG-rich repeat containing 1   \n",
       "NIPAL3                            NIPA like domain containing 3   \n",
       "LAS1L                     LAS1 like, ribosome biogenesis factor   \n",
       "ENPP4         ectonucleotide pyrophosphatase/phosphodiestera...   \n",
       "SEMA3F                                            semaphorin 3F   \n",
       "CFTR          cystic fibrosis transmembrane conductance regu...   \n",
       "ANKIB1               ankyrin repeat and IBR domain containing 1   \n",
       "CYP51A1          cytochrome P450 family 51 subfamily A member 1   \n",
       "KRIT1                          KRIT1, ankyrin repeat containing   \n",
       "RAD52                         RAD52 homolog, DNA repair protein   \n",
       "BAD                       BCL2 associated agonist of cell death   \n",
       "LAP3                                   leucine aminopeptidase 3   \n",
       "CD99                             CD99 molecule (Xg blood group)   \n",
       "HS3ST1         heparan sulfate-glucosamine 3-sulfotransferase 1   \n",
       "AOC1                         amine oxidase, copper containing 1   \n",
       "MAD1L1                     MAD1 mitotic arrest deficient like 1   \n",
       "LASP1                                     LIM and SH3 protein 1   \n",
       "SNX11                                          sorting nexin 11   \n",
       "TMEM176A                             transmembrane protein 176A   \n",
       "M6PR             mannose-6-phosphate receptor, cation dependent   \n",
       "...                                                         ...   \n",
       "AC133065.6                                                  NaN   \n",
       "AC022107.1                                                  NaN   \n",
       "AC005831.1                                                  NaN   \n",
       "AC026401.3                                                  NaN   \n",
       "AC093908.2                                                  NaN   \n",
       "MIR4697HG                                     MIR4697 host gene   \n",
       "AC233723.2                                                  NaN   \n",
       "AP001528.3                                                  NaN   \n",
       "AC000123.3                                                  NaN   \n",
       "Z95331.1                                                    NaN   \n",
       "AP000648.3                                                  NaN   \n",
       "AC245369.3                  Immunoglobulin heavy variable 1-69D   \n",
       "CCDC163                       coiled-coil domain containing 163   \n",
       "SH3PXD2A-AS1                           SH3PXD2A antisense RNA 1   \n",
       "LINC01232           long intergenic non-protein coding RNA 1232   \n",
       "PAGR1                PAXIP1 associated glutamate rich protein 1   \n",
       "LINC00294            long intergenic non-protein coding RNA 294   \n",
       "GSEC            G-quadruplex forming sequence containing lncRNA   \n",
       "FOXCUT           FOXC1 upstream transcript (non-protein coding)   \n",
       "N4BP2L2-IT2                        N4BPL2 intronic transcript 2   \n",
       "TRG-AS1             T-cell receptor gamma locus antisense RNA 1   \n",
       "AC007878.1                                                  NaN   \n",
       "AL133410.3                                                  NaN   \n",
       "AC018645.3                                                  NaN   \n",
       "HELLPAR                    HELLP associated long non-coding RNA   \n",
       "SNHG4                           small nucleolar RNA host gene 4   \n",
       "CICP14          capicua transcriptional repressor pseudogene 14   \n",
       "EBLN3P        endogenous Bornavirus-like nucleoprotein 3, ps...   \n",
       "AP000851.2                                                  NaN   \n",
       "LINC01230           long intergenic non-protein coding RNA 1230   \n",
       "\n",
       "              TCGA-A7-A0CE-01  TCGA-A7-A0CE-11  TCGA-A7-A0CH-01  \\\n",
       "Name                                                              \n",
       "TSPAN6                 8462.0           5404.0           3036.0   \n",
       "TNMD                     54.0            320.0              0.0   \n",
       "DPM1                   2408.0           2472.0           1497.0   \n",
       "SCYL3                  1518.0           1483.0           3030.0   \n",
       "C1orf112               2056.0            312.0            730.0   \n",
       "FGR                     287.0            869.0            173.0   \n",
       "CFH                    1663.0           5760.0           1531.0   \n",
       "FUCA2                 10475.0           4397.0           1772.0   \n",
       "GCLC                   2330.0           2775.0           3522.0   \n",
       "NFYA                   2776.0           2060.0           3629.0   \n",
       "STPG1                  1162.0            845.0            904.0   \n",
       "NIPAL3                 5432.0           4848.0           3319.0   \n",
       "LAS1L                  4739.0           3778.0           2394.0   \n",
       "ENPP4                   946.0           1168.0           2695.0   \n",
       "SEMA3F                 3486.0           5125.0           3911.0   \n",
       "CFTR                      1.0            279.0             33.0   \n",
       "ANKIB1                 7688.0           3182.0           3500.0   \n",
       "CYP51A1                 336.0            227.0            250.0   \n",
       "KRIT1                  3430.0           2505.0           1893.0   \n",
       "RAD52                   930.0            573.0            389.0   \n",
       "BAD                     863.0           2586.0           1004.0   \n",
       "LAP3                   7414.0           9073.0           6335.0   \n",
       "CD99                   5696.0          12627.0           7232.0   \n",
       "HS3ST1                  126.0            347.0            190.0   \n",
       "AOC1                     27.0             53.0             12.0   \n",
       "MAD1L1                 3323.0           1909.0            936.0   \n",
       "LASP1                 13231.0          18957.0          17489.0   \n",
       "SNX11                  1362.0           1942.0           1290.0   \n",
       "TMEM176A               2351.0           5231.0            234.0   \n",
       "M6PR                   9044.0           4161.0           4002.0   \n",
       "...                       ...              ...              ...   \n",
       "AC133065.6               77.0            301.0             56.0   \n",
       "AC022107.1              363.0            336.0            221.0   \n",
       "AC005831.1               19.0            429.0            108.0   \n",
       "AC026401.3              461.0            423.0            307.0   \n",
       "AC093908.2               69.0             87.0            518.0   \n",
       "MIR4697HG                28.0            970.0            228.0   \n",
       "AC233723.2              107.0            387.0            178.0   \n",
       "AP001528.3               46.0            576.0             81.0   \n",
       "AC000123.3              289.0            146.0            255.0   \n",
       "Z95331.1                 21.0             38.0             42.0   \n",
       "AP000648.3              402.0            203.0            184.0   \n",
       "AC245369.3              324.0            658.0             17.0   \n",
       "CCDC163                 299.0            483.0            187.0   \n",
       "SH3PXD2A-AS1             13.0             20.0              4.0   \n",
       "LINC01232               144.0            173.0             96.0   \n",
       "PAGR1                   669.0            418.0            394.0   \n",
       "LINC00294               783.0            724.0            303.0   \n",
       "GSEC                    353.0            144.0             84.0   \n",
       "FOXCUT                  195.0              6.0              1.0   \n",
       "N4BP2L2-IT2              10.0            153.0             79.0   \n",
       "TRG-AS1                  23.0             87.0             18.0   \n",
       "AC007878.1               32.0             16.0             32.0   \n",
       "AL133410.3               81.0            248.0            316.0   \n",
       "AC018645.3              174.0            298.0             98.0   \n",
       "HELLPAR                  75.0            125.0             30.0   \n",
       "SNHG4                   126.0             96.0             66.0   \n",
       "CICP14                  518.0            452.0            247.0   \n",
       "EBLN3P                 3621.0           3397.0           3963.0   \n",
       "AP000851.2                5.0             57.0              6.0   \n",
       "LINC01230                 5.0             11.0              0.0   \n",
       "\n",
       "              TCGA-A7-A0CH-11  TCGA-A7-A0D9-01  TCGA-A7-A0D9-11  \\\n",
       "Name                                                              \n",
       "TSPAN6                 5030.0           2679.0           3616.0   \n",
       "TNMD                   2116.0             73.0           3616.0   \n",
       "DPM1                   1611.0           4135.0           1254.0   \n",
       "SCYL3                  1154.0           1852.0            820.0   \n",
       "C1orf112                252.0            366.0            225.0   \n",
       "FGR                    1536.0            201.0            969.0   \n",
       "CFH                    4695.0           1672.0          14103.0   \n",
       "FUCA2                  3930.0           3733.0           3688.0   \n",
       "GCLC                   3217.0           2527.0           4155.0   \n",
       "NFYA                   1645.0           3048.0           1181.0   \n",
       "STPG1                   855.0            438.0            666.0   \n",
       "NIPAL3                 2896.0           2720.0           1321.0   \n",
       "LAS1L                  2944.0           3330.0           1848.0   \n",
       "ENPP4                  1759.0           2235.0           1971.0   \n",
       "SEMA3F                 3239.0          10121.0           1133.0   \n",
       "CFTR                     56.0             56.0             34.0   \n",
       "ANKIB1                 2429.0           7237.0           3496.0   \n",
       "CYP51A1                 127.0            439.0             72.0   \n",
       "KRIT1                  1484.0           4495.0           1745.0   \n",
       "RAD52                   375.0            654.0            380.0   \n",
       "BAD                    2604.0           1768.0           1378.0   \n",
       "LAP3                   6968.0           4508.0           4866.0   \n",
       "CD99                  20444.0           7911.0          18066.0   \n",
       "HS3ST1                  369.0             81.0            100.0   \n",
       "AOC1                     23.0           1097.0             43.0   \n",
       "MAD1L1                 1696.0           1660.0            979.0   \n",
       "LASP1                 23386.0          13768.0          19853.0   \n",
       "SNX11                  1822.0           1482.0           1613.0   \n",
       "TMEM176A               2252.0           2446.0           1352.0   \n",
       "M6PR                   3960.0           4002.0           3210.0   \n",
       "...                       ...              ...              ...   \n",
       "AC133065.6              229.0             64.0            238.0   \n",
       "AC022107.1              214.0            421.0            128.0   \n",
       "AC005831.1               97.0           1013.0             43.0   \n",
       "AC026401.3              227.0            106.0             88.0   \n",
       "AC093908.2               50.0            114.0             48.0   \n",
       "MIR4697HG               390.0           1141.0             66.0   \n",
       "AC233723.2              325.0            192.0            123.0   \n",
       "AP001528.3             2293.0            145.0           3013.0   \n",
       "AC000123.3              123.0            678.0            192.0   \n",
       "Z95331.1                 23.0             73.0             30.0   \n",
       "AP000648.3              141.0            310.0            187.0   \n",
       "AC245369.3             1898.0             90.0             26.0   \n",
       "CCDC163                 205.0             99.0             29.0   \n",
       "SH3PXD2A-AS1             18.0             26.0             17.0   \n",
       "LINC01232               136.0             66.0            155.0   \n",
       "PAGR1                   419.0            382.0            188.0   \n",
       "LINC00294               818.0            539.0            847.0   \n",
       "GSEC                    132.0            555.0            124.0   \n",
       "FOXCUT                    6.0              8.0              0.0   \n",
       "N4BP2L2-IT2             138.0            115.0             73.0   \n",
       "TRG-AS1                  55.0             20.0             33.0   \n",
       "AC007878.1               24.0             46.0             52.0   \n",
       "AL133410.3              283.0            131.0            175.0   \n",
       "AC018645.3              258.0           1168.0            159.0   \n",
       "HELLPAR                  42.0             46.0             59.0   \n",
       "SNHG4                    48.0             41.0             18.0   \n",
       "CICP14                  380.0            617.0            227.0   \n",
       "EBLN3P                 2696.0           4726.0           2465.0   \n",
       "AP000851.2              121.0              2.0             11.0   \n",
       "LINC01230               109.0             12.0            274.0   \n",
       "\n",
       "              TCGA-A7-A0DB-01  TCGA-A7-A0DB-11  TCGA-A7-A0DC-01  \\\n",
       "Name                                                              \n",
       "TSPAN6                 2203.0           2425.0           4094.0   \n",
       "TNMD                     47.0            304.0             14.0   \n",
       "DPM1                   1131.0           1137.0           3050.0   \n",
       "SCYL3                  1919.0            687.0           1906.0   \n",
       "C1orf112                725.0            241.0            846.0   \n",
       "FGR                     846.0           1517.0            102.0   \n",
       "CFH                    2949.0           8647.0            772.0   \n",
       "FUCA2                  1737.0           3963.0           2124.0   \n",
       "GCLC                   2814.0           2833.0           9453.0   \n",
       "NFYA                   2121.0           1175.0           3308.0   \n",
       "STPG1                   230.0            402.0            277.0   \n",
       "NIPAL3                 1927.0           1200.0           4931.0   \n",
       "LAS1L                   967.0           1470.0           2509.0   \n",
       "ENPP4                  1912.0           1715.0           3074.0   \n",
       "SEMA3F                 2968.0           1729.0           3401.0   \n",
       "CFTR                     24.0             33.0             17.0   \n",
       "ANKIB1                 5265.0           3122.0           9206.0   \n",
       "CYP51A1                 572.0             92.0            882.0   \n",
       "KRIT1                  3068.0           1619.0           3320.0   \n",
       "RAD52                   393.0            271.0            742.0   \n",
       "BAD                     700.0           1099.0            374.0   \n",
       "LAP3                   3474.0           6087.0           5420.0   \n",
       "CD99                   3482.0          22271.0           1572.0   \n",
       "HS3ST1                  189.0            234.0             34.0   \n",
       "AOC1                     63.0             18.0              9.0   \n",
       "MAD1L1                  740.0            940.0           1015.0   \n",
       "LASP1                  5604.0          16959.0          15047.0   \n",
       "SNX11                   640.0           1642.0           1240.0   \n",
       "TMEM176A                192.0           1338.0             50.0   \n",
       "M6PR                   2841.0           3344.0           6640.0   \n",
       "...                       ...              ...              ...   \n",
       "AC133065.6               88.0            147.0             54.0   \n",
       "AC022107.1              121.0            115.0            501.0   \n",
       "AC005831.1               38.0            105.0            260.0   \n",
       "AC026401.3              130.0            169.0             86.0   \n",
       "AC093908.2              240.0             56.0            235.0   \n",
       "MIR4697HG                27.0            250.0             80.0   \n",
       "AC233723.2               63.0             87.0            336.0   \n",
       "AP001528.3              343.0           2212.0            238.0   \n",
       "AC000123.3              346.0            160.0            775.0   \n",
       "Z95331.1                146.0             13.0            314.0   \n",
       "AP000648.3              275.0            159.0            619.0   \n",
       "AC245369.3               11.0             15.0              1.0   \n",
       "CCDC163                  43.0             83.0            178.0   \n",
       "SH3PXD2A-AS1             35.0             15.0             58.0   \n",
       "LINC01232               114.0            209.0             58.0   \n",
       "PAGR1                   378.0            167.0            773.0   \n",
       "LINC00294               200.0            806.0            421.0   \n",
       "GSEC                     70.0            108.0             63.0   \n",
       "FOXCUT                    1.0              3.0              0.0   \n",
       "N4BP2L2-IT2             369.0             94.0            713.0   \n",
       "TRG-AS1                  40.0             78.0             22.0   \n",
       "AC007878.1              355.0             33.0            514.0   \n",
       "AL133410.3              203.0             82.0             48.0   \n",
       "AC018645.3               34.0            178.0             28.0   \n",
       "HELLPAR                 200.0             74.0            681.0   \n",
       "SNHG4                    82.0             16.0            121.0   \n",
       "CICP14                  358.0            357.0            358.0   \n",
       "EBLN3P                 4153.0           2378.0           8902.0   \n",
       "AP000851.2                2.0              4.0              0.0   \n",
       "LINC01230                14.0            820.0              0.0   \n",
       "\n",
       "                   ...         TCGA-E9-A1RF-01  TCGA-E9-A1RF-11  \\\n",
       "Name               ...                                            \n",
       "TSPAN6             ...                   676.0           4544.0   \n",
       "TNMD               ...                    41.0           1881.0   \n",
       "DPM1               ...                  3732.0           1565.0   \n",
       "SCYL3              ...                  3155.0           1356.0   \n",
       "C1orf112           ...                  1010.0            294.0   \n",
       "FGR                ...                   910.0           1006.0   \n",
       "CFH                ...                  5382.0          24121.0   \n",
       "FUCA2              ...                  3559.0           3695.0   \n",
       "GCLC               ...                  2968.0           5097.0   \n",
       "NFYA               ...                  3704.0           2025.0   \n",
       "STPG1              ...                   704.0            591.0   \n",
       "NIPAL3             ...                  2479.0           1388.0   \n",
       "LAS1L              ...                  5322.0           1665.0   \n",
       "ENPP4              ...                  1223.0           3793.0   \n",
       "SEMA3F             ...                  7643.0           1205.0   \n",
       "CFTR               ...                     3.0             62.0   \n",
       "ANKIB1             ...                  3608.0           5260.0   \n",
       "CYP51A1            ...                   149.0             95.0   \n",
       "KRIT1              ...                  1403.0           2650.0   \n",
       "RAD52              ...                   562.0            590.0   \n",
       "BAD                ...                  1738.0            807.0   \n",
       "LAP3               ...                 11987.0           5799.0   \n",
       "CD99               ...                 15759.0          18304.0   \n",
       "HS3ST1             ...                   181.0            158.0   \n",
       "AOC1               ...                    64.0              7.0   \n",
       "MAD1L1             ...                  1164.0            456.0   \n",
       "LASP1              ...                 26291.0          13435.0   \n",
       "SNX11              ...                  3997.0           2195.0   \n",
       "TMEM176A           ...                  5008.0           1659.0   \n",
       "M6PR               ...                  5028.0           4226.0   \n",
       "...                ...                     ...              ...   \n",
       "AC133065.6         ...                   160.0            296.0   \n",
       "AC022107.1         ...                   158.0             91.0   \n",
       "AC005831.1         ...                   465.0            152.0   \n",
       "AC026401.3         ...                   342.0            110.0   \n",
       "AC093908.2         ...                   169.0             73.0   \n",
       "MIR4697HG          ...                    16.0            117.0   \n",
       "AC233723.2         ...                   179.0             92.0   \n",
       "AP001528.3         ...                    85.0           5370.0   \n",
       "AC000123.3         ...                   202.0            427.0   \n",
       "Z95331.1           ...                    36.0             47.0   \n",
       "AP000648.3         ...                   262.0            383.0   \n",
       "AC245369.3         ...                  1386.0              5.0   \n",
       "CCDC163            ...                   251.0            104.0   \n",
       "SH3PXD2A-AS1       ...                     9.0             11.0   \n",
       "LINC01232          ...                   149.0            229.0   \n",
       "PAGR1              ...                   262.0             69.0   \n",
       "LINC00294          ...                   781.0           1423.0   \n",
       "GSEC               ...                    98.0            124.0   \n",
       "FOXCUT             ...                     0.0              0.0   \n",
       "N4BP2L2-IT2        ...                   277.0            213.0   \n",
       "TRG-AS1            ...                   128.0             53.0   \n",
       "AC007878.1         ...                    59.0            128.0   \n",
       "AL133410.3         ...                   107.0            153.0   \n",
       "AC018645.3         ...                   144.0            203.0   \n",
       "HELLPAR            ...                    79.0            100.0   \n",
       "SNHG4              ...                   237.0             30.0   \n",
       "CICP14             ...                   329.0            268.0   \n",
       "EBLN3P             ...                  3758.0           3907.0   \n",
       "AP000851.2         ...                     8.0             12.0   \n",
       "LINC01230          ...                     3.0            418.0   \n",
       "\n",
       "              TCGA-E9-A1RH-01  TCGA-E9-A1RH-11  TCGA-E9-A1RI-01  \\\n",
       "Name                                                              \n",
       "TSPAN6                  327.0           3581.0           1980.0   \n",
       "TNMD                      0.0            770.0              5.0   \n",
       "DPM1                   6267.0           1615.0           2885.0   \n",
       "SCYL3                   775.0           1216.0           2538.0   \n",
       "C1orf112                770.0            236.0            741.0   \n",
       "FGR                     338.0            578.0            399.0   \n",
       "CFH                    2915.0          13543.0           4207.0   \n",
       "FUCA2                  2814.0           3660.0           2639.0   \n",
       "GCLC                   4311.0           3845.0           3539.0   \n",
       "NFYA                   5660.0           1556.0           3410.0   \n",
       "STPG1                   328.0            590.0            913.0   \n",
       "NIPAL3                 1356.0           2201.0           3485.0   \n",
       "LAS1L                  8071.0           2031.0           3179.0   \n",
       "ENPP4                   182.0           2605.0           1450.0   \n",
       "SEMA3F                 2847.0           2061.0           5777.0   \n",
       "CFTR                      1.0             26.0             20.0   \n",
       "ANKIB1                10522.0           4264.0           8070.0   \n",
       "CYP51A1                 256.0            109.0            204.0   \n",
       "KRIT1                  2409.0           2316.0           3391.0   \n",
       "RAD52                   365.0            429.0            657.0   \n",
       "BAD                     835.0           1360.0           1526.0   \n",
       "LAP3                   6567.0           6457.0           8764.0   \n",
       "CD99                  11117.0          16453.0          13723.0   \n",
       "HS3ST1                  240.0            264.0            160.0   \n",
       "AOC1                     22.0             19.0             10.0   \n",
       "MAD1L1                 1452.0            860.0            851.0   \n",
       "LASP1                 31483.0          14566.0          32197.0   \n",
       "SNX11                  1580.0           1997.0           2927.0   \n",
       "TMEM176A                487.0           2157.0           1641.0   \n",
       "M6PR                   4758.0           3436.0           5041.0   \n",
       "...                       ...              ...              ...   \n",
       "AC133065.6              178.0            256.0            185.0   \n",
       "AC022107.1              496.0            137.0            439.0   \n",
       "AC005831.1               38.0            137.0            652.0   \n",
       "AC026401.3              600.0            121.0            312.0   \n",
       "AC093908.2             2353.0             50.0            346.0   \n",
       "MIR4697HG                 2.0            259.0             79.0   \n",
       "AC233723.2              340.0            126.0            144.0   \n",
       "AP001528.3               55.0           3617.0            283.0   \n",
       "AC000123.3              193.0            263.0            323.0   \n",
       "Z95331.1                 51.0             36.0             37.0   \n",
       "AP000648.3              100.0            237.0            201.0   \n",
       "AC245369.3             1385.0            194.0            708.0   \n",
       "CCDC163                 137.0             42.0            483.0   \n",
       "SH3PXD2A-AS1             27.0             17.0              9.0   \n",
       "LINC01232              1060.0            219.0            215.0   \n",
       "PAGR1                   449.0            125.0            302.0   \n",
       "LINC00294               772.0            943.0            791.0   \n",
       "GSEC                     93.0            104.0            196.0   \n",
       "FOXCUT                    0.0              3.0              0.0   \n",
       "N4BP2L2-IT2              39.0            158.0            246.0   \n",
       "TRG-AS1                   7.0             51.0             63.0   \n",
       "AC007878.1               12.0             28.0             61.0   \n",
       "AL133410.3               22.0            270.0            158.0   \n",
       "AC018645.3              148.0            274.0            300.0   \n",
       "HELLPAR                  39.0            111.0             86.0   \n",
       "SNHG4                    29.0             39.0             56.0   \n",
       "CICP14                  156.0            224.0            238.0   \n",
       "EBLN3P                 3538.0           3913.0           9734.0   \n",
       "AP000851.2                0.0             48.0             30.0   \n",
       "LINC01230                 1.0            613.0              2.0   \n",
       "\n",
       "              TCGA-E9-A1RI-11  TCGA-GI-A2C8-01  TCGA-GI-A2C8-11  \\\n",
       "Name                                                              \n",
       "TSPAN6                 7639.0           8586.0            422.0   \n",
       "TNMD                   3541.0              5.0             16.0   \n",
       "DPM1                   1999.0           2687.0           1675.0   \n",
       "SCYL3                  1287.0           3240.0            751.0   \n",
       "C1orf112                290.0            617.0             88.0   \n",
       "FGR                     377.0            475.0            129.0   \n",
       "CFH                   19118.0           4565.0           3231.0   \n",
       "FUCA2                  3746.0           5589.0            726.0   \n",
       "GCLC                   7854.0           3397.0           1244.0   \n",
       "NFYA                   1776.0           1798.0            938.0   \n",
       "STPG1                   723.0            556.0            138.0   \n",
       "NIPAL3                 1850.0           2534.0            889.0   \n",
       "LAS1L                  1638.0           2738.0            593.0   \n",
       "ENPP4                  4956.0           6140.0           3424.0   \n",
       "SEMA3F                 1941.0           6497.0            553.0   \n",
       "CFTR                     21.0             59.0              0.0   \n",
       "ANKIB1                 6739.0           6016.0           2298.0   \n",
       "CYP51A1                  89.0            326.0             83.0   \n",
       "KRIT1                  2540.0           2427.0           1729.0   \n",
       "RAD52                   467.0            439.0            498.0   \n",
       "BAD                     768.0           1500.0            642.0   \n",
       "LAP3                   6895.0           4841.0           4281.0   \n",
       "CD99                  17053.0          11737.0           5089.0   \n",
       "HS3ST1                  105.0            252.0             55.0   \n",
       "AOC1                     15.0             57.0              3.0   \n",
       "MAD1L1                  523.0           1314.0            805.0   \n",
       "LASP1                 18844.0          13011.0           3013.0   \n",
       "SNX11                  2145.0           1949.0            755.0   \n",
       "TMEM176A               2614.0          13815.0            572.0   \n",
       "M6PR                   3540.0           4361.0           1764.0   \n",
       "...                       ...              ...              ...   \n",
       "AC133065.6              242.0            301.0             26.0   \n",
       "AC022107.1              178.0            292.0             30.0   \n",
       "AC005831.1              106.0            534.0             20.0   \n",
       "AC026401.3               96.0            123.0             46.0   \n",
       "AC093908.2              150.0            546.0             20.0   \n",
       "MIR4697HG                55.0            889.0            305.0   \n",
       "AC233723.2              120.0            174.0             83.0   \n",
       "AP001528.3             7830.0            140.0            368.0   \n",
       "AC000123.3              453.0            614.0             97.0   \n",
       "Z95331.1                 51.0             28.0              9.0   \n",
       "AP000648.3              180.0            120.0            126.0   \n",
       "AC245369.3               28.0             99.0              0.0   \n",
       "CCDC163                 217.0            339.0             66.0   \n",
       "SH3PXD2A-AS1             14.0             19.0              2.0   \n",
       "LINC01232               196.0             73.0             49.0   \n",
       "PAGR1                    76.0            455.0            176.0   \n",
       "LINC00294              1121.0           3464.0            221.0   \n",
       "GSEC                    145.0            573.0             19.0   \n",
       "FOXCUT                    1.0              0.0              0.0   \n",
       "N4BP2L2-IT2             188.0            137.0             13.0   \n",
       "TRG-AS1                  44.0             54.0              5.0   \n",
       "AC007878.1               77.0             42.0             14.0   \n",
       "AL133410.3              299.0           1301.0            337.0   \n",
       "AC018645.3              256.0            280.0            136.0   \n",
       "HELLPAR                 102.0             26.0             52.0   \n",
       "SNHG4                    29.0            164.0             38.0   \n",
       "CICP14                  219.0            410.0            121.0   \n",
       "EBLN3P                 4670.0           5983.0           1940.0   \n",
       "AP000851.2                8.0             15.0              1.0   \n",
       "LINC01230               329.0              1.0              5.0   \n",
       "\n",
       "              TCGA-GI-A2C9-01  TCGA-GI-A2C9-11  \n",
       "Name                                            \n",
       "TSPAN6                 3944.0           4849.0  \n",
       "TNMD                      2.0            310.0  \n",
       "DPM1                   2708.0           2290.0  \n",
       "SCYL3                  1524.0           1265.0  \n",
       "C1orf112                657.0            343.0  \n",
       "FGR                     422.0            969.0  \n",
       "CFH                    2747.0          22260.0  \n",
       "FUCA2                  4024.0           3422.0  \n",
       "GCLC                   2673.0           2536.0  \n",
       "NFYA                   4720.0           3317.0  \n",
       "STPG1                  1418.0            922.0  \n",
       "NIPAL3                 7065.0           5090.0  \n",
       "LAS1L                  3153.0           2688.0  \n",
       "ENPP4                  1753.0           2302.0  \n",
       "SEMA3F                 1331.0           3759.0  \n",
       "CFTR                     14.0             55.0  \n",
       "ANKIB1                 6059.0           4562.0  \n",
       "CYP51A1                 378.0            426.0  \n",
       "KRIT1                  2351.0           2927.0  \n",
       "RAD52                   872.0            843.0  \n",
       "BAD                    2554.0           1731.0  \n",
       "LAP3                   6757.0           7810.0  \n",
       "CD99                   8724.0          12885.0  \n",
       "HS3ST1                 1301.0            510.0  \n",
       "AOC1                    161.0             54.0  \n",
       "MAD1L1                 1432.0           1631.0  \n",
       "LASP1                 10706.0          16794.0  \n",
       "SNX11                  2019.0           1447.0  \n",
       "TMEM176A                939.0           3105.0  \n",
       "M6PR                  12882.0           5210.0  \n",
       "...                       ...              ...  \n",
       "AC133065.6              129.0            356.0  \n",
       "AC022107.1               77.0            148.0  \n",
       "AC005831.1              177.0            355.0  \n",
       "AC026401.3              798.0            437.0  \n",
       "AC093908.2               45.0            102.0  \n",
       "MIR4697HG                34.0            680.0  \n",
       "AC233723.2              106.0            254.0  \n",
       "AP001528.3               29.0            827.0  \n",
       "AC000123.3              190.0            316.0  \n",
       "Z95331.1                 77.0             50.0  \n",
       "AP000648.3              344.0            331.0  \n",
       "AC245369.3               16.0            488.0  \n",
       "CCDC163                 388.0            210.0  \n",
       "SH3PXD2A-AS1             68.0             19.0  \n",
       "LINC01232               160.0            277.0  \n",
       "PAGR1                   404.0            496.0  \n",
       "LINC00294              1420.0            950.0  \n",
       "GSEC                   1074.0            128.0  \n",
       "FOXCUT                   66.0             12.0  \n",
       "N4BP2L2-IT2             149.0            187.0  \n",
       "TRG-AS1                  40.0            106.0  \n",
       "AC007878.1               25.0            162.0  \n",
       "AL133410.3              102.0            303.0  \n",
       "AC018645.3               67.0            201.0  \n",
       "HELLPAR                  64.0             98.0  \n",
       "SNHG4                   222.0            226.0  \n",
       "CICP14                  260.0            691.0  \n",
       "EBLN3P                 4337.0           4834.0  \n",
       "AP000851.2             1555.0            130.0  \n",
       "LINC01230                 0.0             29.0  \n",
       "\n",
       "[16339 rows x 225 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "std = df.std(axis=1)\n",
    "min_std = 100\n",
    "# mean = df.mean(axis=1)\n",
    "new_std = std[std > min_std]\n",
    "print(\"{} genes were filtered out -- they had a Standard Deviation lower than {} (mean std was ~{})\".format(std.count() - new_std.count(), min_std, int(std.mean()) ))\n",
    "print(\"New gene lengths is {}\".format(new_std.count()))\n",
    "filtered_df = df.ix[new_std.index]\n",
    "filtered_df.insert(loc=0,column='Description', value = Description.ix[new_std.index])\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import cuzcatlan as cusca\n",
    "cusca.df2gct(filtered_df, name='TCGA_BRCA_all_matched_var_filtered.gct')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "genepattern": {
     "server": "https://gp-beta-ami.genepattern.org/gp",
     "type": "auth"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "12e30ba87fb1432489ec9c70ef7693f8",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPAuthWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPAuthWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Requires GenePattern Notebook: pip install genepattern-notebook\n",
    "import gp\n",
    "import genepattern\n",
    "\n",
    "# Username and password removed for security reasons.\n",
    "genepattern.GPAuthWidget(genepattern.register_session(\"https://gp-beta-ami.genepattern.org/gp\", \"\", \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "beeff5153b5945ca8bd62f2003b6e251",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "renamefile_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338')\n",
    "renamefile_job_spec = renamefile_task.make_job_spec()\n",
    "renamefile_job_spec.set_parameter(\"input.file\", \"https://datasets.genepattern.org/data/TCGA_BRCA_temp/TCGA_BRCA_all_matched_samples.gct\")\n",
    "renamefile_job_spec.set_parameter(\"output.filename\", \"BRCA.gct\")\n",
    "renamefile_job_spec.set_parameter(\"screen.filename\", \"no\")\n",
    "renamefile_job_spec.set_parameter(\"force.copy\", \"no\")\n",
    "genepattern.GPTaskWidget(renamefile_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f10852f99df14201a1ae7414e03cce36",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=20823)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job20823 = gp.GPJob(genepattern.get_session(0), 20823)\n",
    "genepattern.GPJobWidget(job20823)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preprocess Dataset "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3e1c18f43f9d4f79aa4aaf507429d800",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "preprocessdataset_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020')\n",
    "preprocessdataset_job_spec = preprocessdataset_task.make_job_spec()\n",
    "preprocessdataset_job_spec.set_parameter(\"input.filename\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18035/BRCA.gct\")\n",
    "preprocessdataset_job_spec.set_parameter(\"threshold.and.filter\", \"1\")\n",
    "preprocessdataset_job_spec.set_parameter(\"floor\", \"20\")\n",
    "preprocessdataset_job_spec.set_parameter(\"ceiling\", \"2000000\")\n",
    "preprocessdataset_job_spec.set_parameter(\"min.fold.change\", \"3\")\n",
    "preprocessdataset_job_spec.set_parameter(\"min.delta\", \"100\")\n",
    "preprocessdataset_job_spec.set_parameter(\"num.outliers.to.exclude\", \"0\")\n",
    "preprocessdataset_job_spec.set_parameter(\"row.normalization\", \"0\")\n",
    "preprocessdataset_job_spec.set_parameter(\"row.sampling.rate\", \"1\")\n",
    "preprocessdataset_job_spec.set_parameter(\"threshold.for.removing.rows\", \"\")\n",
    "preprocessdataset_job_spec.set_parameter(\"number.of.columns.above.threshold\", \"\")\n",
    "preprocessdataset_job_spec.set_parameter(\"log2.transform\", \"0\")\n",
    "preprocessdataset_job_spec.set_parameter(\"output.file.format\", \"3\")\n",
    "preprocessdataset_job_spec.set_parameter(\"output.file\", \"<input.filename_basename>.preprocessed\")\n",
    "genepattern.GPTaskWidget(preprocessdataset_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b707aedb0f3b497e81011eb9f997c51f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18036)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18036 = gp.GPJob(genepattern.get_session(0), 18036)\n",
    "genepattern.GPJobWidget(job18036)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5a7d4cd567ea41c8b97ac1855a19e572",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hierarchicalclustering_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009')\n",
    "hierarchicalclustering_job_spec = hierarchicalclustering_task.make_job_spec()\n",
    "hierarchicalclustering_job_spec.set_parameter(\"input.filename\", \"https://datasets.genepattern.org/data/TCGA_BRCA_temp/TCGA_BRCA_all_matched_samples.gct\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"column.distance.measure\", \"2\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"output_distance_matrix\", \"False\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.distance.measure\", \"0\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"clustering.method\", \"a\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"output.base.name\", \"<input.filename_basename>\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.centering\", \"Mean\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.normalize\", \"False\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"col.centering\", \"Mean\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"col.normalize\", \"False\")\n",
    "genepattern.GPTaskWidget(hierarchicalclustering_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f312665c658340c1bbba71c8836eb1f4",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18044)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18044 = gp.GPJob(genepattern.get_session(0), 18044)\n",
    "genepattern.GPJobWidget(job18044)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c033d255253f4f7f89d4cfddd82e8dc0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hierarchicalclusteringviewer_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031')\n",
    "hierarchicalclusteringviewer_job_spec = hierarchicalclusteringviewer_task.make_job_spec()\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"cdt.file\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18044/TCGA_BRCA_all_matched_samples.cdt\")\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"gtr.file\", \"\")\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"atr.file\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18044/TCGA_BRCA_all_matched_samples.atr\")\n",
    "genepattern.GPTaskWidget(hierarchicalclusteringviewer_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "70dff75728504517a1a35a19b9fefe42",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18348)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18348 = gp.GPJob(genepattern.get_session(0), 18348)\n",
    "genepattern.GPJobWidget(job18348)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6db55336696d4201b2c8532068f6697b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hierarchicalclustering_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009')\n",
    "hierarchicalclustering_job_spec = hierarchicalclustering_task.make_job_spec()\n",
    "hierarchicalclustering_job_spec.set_parameter(\"input.filename\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18036/BRCA.preprocessed.gct\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"column.distance.measure\", \"2\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"output_distance_matrix\", \"False\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.distance.measure\", \"0\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"clustering.method\", \"a\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"output.base.name\", \"<input.filename_basename>\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.centering\", \"Mean\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"row.normalize\", \"True\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"col.centering\", \"Mean\")\n",
    "hierarchicalclustering_job_spec.set_parameter(\"col.normalize\", \"True\")\n",
    "genepattern.GPTaskWidget(hierarchicalclustering_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f6b56ece779b40399dbc3c9fbb734156",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18043)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18043 = gp.GPJob(genepattern.get_session(0), 18043)\n",
    "genepattern.GPJobWidget(job18043)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b658f876c10b482f9da22e2ef5e6ea06",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hierarchicalclusteringviewer_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031')\n",
    "hierarchicalclusteringviewer_job_spec = hierarchicalclusteringviewer_task.make_job_spec()\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"cdt.file\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18043/BRCA.preprocessed.cdt\")\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"gtr.file\", \"\")\n",
    "hierarchicalclusteringviewer_job_spec.set_parameter(\"atr.file\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18043/BRCA.preprocessed.atr\")\n",
    "genepattern.GPTaskWidget(hierarchicalclusteringviewer_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2637a139b3b5435794f827d5d6d90480",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18046)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18046 = gp.GPJob(genepattern.get_session(0), 18046)\n",
    "genepattern.GPJobWidget(job18046)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "genepattern": {
     "param_values": {
      "clustering_type": "Single",
      "clusters_to_highlight": "6",
      "file_basename": "HC_out_2",
      "input_gene_expression": [
       "https://datasets.genepattern.org/data/TCGA_BRCA_temp/TCGA_BRCA_all_matched.filtered.gct"
      ]
     },
     "show_code": false,
     "type": "uibuilder"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4d52d276004d4aeb8717b1ff15fb0308",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPUIBuilder</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPUIBuilder(description='This function performs hierarchical clustering to group samples (columns) with similar phenotypes..', function_import='hc_samples', name='Hierarchical Clustering of Samples (Columns).', params=[{'name': 'input_gene_expression', 'label': 'input_gene_expression', 'optional': False, 'default': '', 'description': 'gene expression data filename (.gct file) where rows are genes and columns are samples', 'hide': False, 'type': 'file', 'kinds': ['gct'], 'choices': []}, {'name': 'clustering_type', 'label': 'clustering_type', 'optional': False, 'default': '', 'description': 'single or consensus -- Only single is suported at the moment', 'hide': False, 'type': 'text', 'kinds': None, 'choices': []}, {'name': 'distance_metric', 'label': 'distance_metric', 'optional': False, 'default': 'pearson', 'description': 'the function to be used when comparing the distance/similarity of the columns in the input_gene_expression dataset', 'hide': False, 'type': 'text', 'kinds': None, 'choices': {'Information Coefficient': 'information_coefficient', 'City Block (Manhattan or L1-norm)': 'manhattan', 'Euclidean (L2-norm)': 'euclidean', 'Pearson Correlation': 'pearson', 'Uncentered Pearson Correlation': 'uncentered_pearson', 'Uncentered Pearson Correlation, absolute value': 'absolute_uncentered_pearson', 'Spearman Correlation': 'spearman', \"Kendall's Tau\": 'kendall', 'Cosine distance': 'cosine'}}, {'name': 'file_basename', 'label': 'file_basename', 'optional': True, 'default': 'HC_out', 'description': 'the name to use when naming output files', 'hide': False, 'type': 'text', 'kinds': None, 'choices': []}, {'name': 'clusters_to_highlight', 'label': 'clusters_to_highlight', 'optional': True, 'default': 'None', 'description': 'how many clusters to highlight in the dendrogram', 'hide': False, 'type': 'text', 'kinds': None, 'choices': []}])"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import genepattern\n",
    "# import cuzcatlan as cusca\n",
    "# import pandas as pd    \n",
    "from cuzcatlan import hc_samples\n",
    "\n",
    "genepattern.GPUIBuilder(hc_samples, name=\"Hierarchical Clustering of Samples (Columns).\", \n",
    "                        description=\"This function performs hierarchical clustering to group samples (columns) with similar phenotypes..\",\n",
    "                        parameters={\n",
    "                                \"distance_metric\":{\n",
    "                                                      \"default\": \"pearson\",\n",
    "                                                      \"choices\":{'Information Coefficient':\"information_coefficient\",\n",
    "                                                                 'City Block (Manhattan or L1-norm)':'manhattan',\n",
    "                                                                 'Euclidean (L2-norm)':\"euclidean\",\n",
    "                                                                 'Pearson Correlation':\"pearson\",\n",
    "                                                                 'Uncentered Pearson Correlation':'uncentered_pearson',\n",
    "                                                                 'Uncentered Pearson Correlation, absolute value':'absolute_uncentered_pearson',\n",
    "                                                                 'Spearman Correlation':'spearman',\n",
    "                                                                 \"Kendall's Tau\": 'kendall',\n",
    "                                                                 'Cosine distance':'cosine',\n",
    "                                                                }\n",
    "                                                     },\n",
    "                            \"input_gene_expression\":{\n",
    "                                \"type\": \"file\",\n",
    "                                \"kinds\":[\"gct\"]}\n",
    "                        })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currenty clustering_type is being ignored, only 'single' is supported.\n",
      "Now we will start performing hierarchical clustering, this may take a little while.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-25-49ce0194d126>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mhc_samples\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_gene_expression\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"https://datasets.genepattern.org/data/TCGA_BRCA_temp/TCGA_BRCA_all_matched.filtered.gct\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mclustering_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"Single\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdistance_metric\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"pearson\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile_basename\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"HC_out_2\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mclusters_to_highlight\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m6\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/jovyan/.local/lib/python3.6/site-packages/cuzcatlan/clustering.py\u001b[0m in \u001b[0;36mhc_samples\u001b[0;34m(input_gene_expression, clustering_type, distance_metric, file_basename, clusters_to_highlight)\u001b[0m\n\u001b[1;32m   1665\u001b[0m                                                   \u001b[0mcustom_plot\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1666\u001b[0m                                                   \u001b[0mclusters_to_highlight\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1667\u001b[0;31m                                                   show)\n\u001b[0m\u001b[1;32m   1668\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Done with Hierarchical Clustering!\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1669\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/jovyan/.local/lib/python3.6/site-packages/cuzcatlan/clustering.py\u001b[0m in \u001b[0;36mHierarchicalClustering\u001b[0;34m(pwd, gct_name, col_distance_metric, row_distance_metric, clustering_method, output_base_name, row_normalization, col_normalization, row_centering, col_centering, output_distances, custom_plot, clusters_to_highlight, show)\u001b[0m\n\u001b[1;32m   1473\u001b[0m         \u001b[0;34m[\u001b[0m\u001b[0mlabel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_rotation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m90\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mlabel\u001b[0m \u001b[0;32min\u001b[0m \u001b[0max1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_xticklabels\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1474\u001b[0m         \u001b[0mfile_path_plot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0moutput_base_name\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'.pdf'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1475\u001b[0;31m         \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msavefig\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile_path_plot\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbbox_inches\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'tight'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1476\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1477\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"----------------------------------------------------------------------\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/pyplot.py\u001b[0m in \u001b[0;36msavefig\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    708\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0msavefig\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    709\u001b[0m     \u001b[0mfig\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 710\u001b[0;31m     \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msavefig\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    711\u001b[0m     \u001b[0mfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcanvas\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw_idle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m   \u001b[0;31m# need this if 'transparent=True' to reset colors\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    712\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/figure.py\u001b[0m in \u001b[0;36msavefig\u001b[0;34m(self, fname, **kwargs)\u001b[0m\n\u001b[1;32m   2033\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_frameon\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mframeon\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2034\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2035\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcanvas\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprint_figure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2036\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2037\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mframeon\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backend_bases.py\u001b[0m in \u001b[0;36mprint_figure\u001b[0;34m(self, filename, dpi, facecolor, edgecolor, orientation, format, **kwargs)\u001b[0m\n\u001b[1;32m   2208\u001b[0m                     \u001b[0morientation\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morientation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2209\u001b[0m                     \u001b[0mdryrun\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2210\u001b[0;31m                     **kwargs)\n\u001b[0m\u001b[1;32m   2211\u001b[0m                 \u001b[0mrenderer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfigure\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cachedRenderer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2212\u001b[0m                 \u001b[0mbbox_inches\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfigure\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_tightbbox\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backends/backend_pdf.py\u001b[0m in \u001b[0;36mprint_pdf\u001b[0;34m(self, filename, **kwargs)\u001b[0m\n\u001b[1;32m   2584\u001b[0m                 \u001b[0mRendererPdf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mimage_dpi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheight\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwidth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2585\u001b[0m                 bbox_inches_restore=_bbox_inches_restore)\n\u001b[0;32m-> 2586\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfigure\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2587\u001b[0m             \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfinalize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2588\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mPdfPages\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/artist.py\u001b[0m in \u001b[0;36mdraw_wrapper\u001b[0;34m(artist, renderer, *args, **kwargs)\u001b[0m\n\u001b[1;32m     53\u001b[0m                 \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0martist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0martist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_agg_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/figure.py\u001b[0m in \u001b[0;36mdraw\u001b[0;34m(self, renderer)\u001b[0m\n\u001b[1;32m   1473\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1474\u001b[0m             mimage._draw_list_compositing_images(\n\u001b[0;32m-> 1475\u001b[0;31m                 renderer, self, artists, self.suppressComposite)\n\u001b[0m\u001b[1;32m   1476\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1477\u001b[0m             \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose_group\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'figure'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/image.py\u001b[0m in \u001b[0;36m_draw_list_compositing_images\u001b[0;34m(renderer, parent, artists, suppress_composite)\u001b[0m\n\u001b[1;32m    139\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mnot_composite\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mhas_images\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0ma\u001b[0m \u001b[0;32min\u001b[0m \u001b[0martists\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m             \u001b[0ma\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m         \u001b[0;31m# Composite any adjacent images together\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/artist.py\u001b[0m in \u001b[0;36mdraw_wrapper\u001b[0;34m(artist, renderer, *args, **kwargs)\u001b[0m\n\u001b[1;32m     53\u001b[0m                 \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0martist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0martist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_agg_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/axes/_base.py\u001b[0m in \u001b[0;36mdraw\u001b[0;34m(self, renderer, inframe)\u001b[0m\n\u001b[1;32m   2605\u001b[0m             \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstop_rasterizing\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2606\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2607\u001b[0;31m         \u001b[0mmimage\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_draw_list_compositing_images\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0martists\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2608\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2609\u001b[0m         \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose_group\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'axes'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/image.py\u001b[0m in \u001b[0;36m_draw_list_compositing_images\u001b[0;34m(renderer, parent, artists, suppress_composite)\u001b[0m\n\u001b[1;32m    139\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mnot_composite\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mhas_images\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0ma\u001b[0m \u001b[0;32min\u001b[0m \u001b[0martists\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m             \u001b[0ma\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m         \u001b[0;31m# Composite any adjacent images together\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/artist.py\u001b[0m in \u001b[0;36mdraw_wrapper\u001b[0;34m(artist, renderer, *args, **kwargs)\u001b[0m\n\u001b[1;32m     53\u001b[0m                 \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0martist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0martist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_agg_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/collections.py\u001b[0m in \u001b[0;36mdraw\u001b[0;34m(self, renderer)\u001b[0m\n\u001b[1;32m   1980\u001b[0m                 \u001b[0mgc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransform\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrozen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_meshWidth\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_meshHeight\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1981\u001b[0m                 \u001b[0mcoordinates\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moffsets\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransOffset\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_facecolor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1982\u001b[0;31m                 self._antialiased, self.get_edgecolors())\n\u001b[0m\u001b[1;32m   1983\u001b[0m         \u001b[0mgc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrestore\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1984\u001b[0m         \u001b[0mrenderer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose_group\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backend_bases.py\u001b[0m in \u001b[0;36mdraw_quad_mesh\u001b[0;34m(self, gc, master_transform, meshWidth, meshHeight, coordinates, offsets, offsetTrans, facecolors, antialiased, edgecolors)\u001b[0m\n\u001b[1;32m    376\u001b[0m         return self.draw_path_collection(\n\u001b[1;32m    377\u001b[0m             \u001b[0mgc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmaster_transform\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpaths\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moffsets\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moffsetTrans\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfacecolors\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 378\u001b[0;31m             edgecolors, linewidths, [], [antialiased], [None], 'screen')\n\u001b[0m\u001b[1;32m    379\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    380\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mdraw_gouraud_triangle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpoints\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolors\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransform\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backends/backend_pdf.py\u001b[0m in \u001b[0;36mdraw_path_collection\u001b[0;34m(self, gc, master_transform, paths, all_transforms, offsets, offsetTrans, facecolors, edgecolors, linewidths, linestyles, antialiaseds, urls, offset_position)\u001b[0m\n\u001b[1;32m   1756\u001b[0m                 \u001b[0moffsets\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moffsetTrans\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfacecolors\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0medgecolors\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1757\u001b[0m                 \u001b[0mlinewidths\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinestyles\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mantialiaseds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murls\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1758\u001b[0;31m                 offset_position)\n\u001b[0m\u001b[1;32m   1759\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1760\u001b[0m         \u001b[0mpadding\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlinewidths\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backend_bases.py\u001b[0m in \u001b[0;36mdraw_path_collection\u001b[0;34m(self, gc, master_transform, paths, all_transforms, offsets, offsetTrans, facecolors, edgecolors, linewidths, linestyles, antialiaseds, urls, offset_position)\u001b[0m\n\u001b[1;32m    355\u001b[0m             transform = transforms.Affine2D(\n\u001b[1;32m    356\u001b[0m                             transform.get_matrix()).translate(xo, yo)\n\u001b[0;32m--> 357\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw_path\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgc0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransform\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrgbFace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    358\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    359\u001b[0m     def draw_quad_mesh(self, gc, master_transform, meshWidth, meshHeight,\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backends/backend_pdf.py\u001b[0m in \u001b[0;36mdraw_path\u001b[0;34m(self, gc, path, transform, rgbFace)\u001b[0m\n\u001b[1;32m   1709\u001b[0m             \u001b[0mrgbFace\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mgc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_hatch_path\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1710\u001b[0m             gc.get_sketch_params())\n\u001b[0;32m-> 1711\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpaint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1712\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1713\u001b[0m     def draw_path_collection(self, gc, master_transform, paths, all_transforms,\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backends/backend_pdf.py\u001b[0m in \u001b[0;36mpaint\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   2252\u001b[0m         \u001b[0mstroked\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilled\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mboth\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2253\u001b[0m         \"\"\"\n\u001b[0;32m-> 2254\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mOp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpaint_path\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfill\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstroke\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2255\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2256\u001b[0m     \u001b[0mcapstyles\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'butt'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'round'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'projecting'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/envs/python3.6/lib/python3.6/site-packages/matplotlib/backends/backend_pdf.py\u001b[0m in \u001b[0;36mfill\u001b[0;34m(self, *args)\u001b[0m\n\u001b[1;32m   2243\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2244\u001b[0m             \u001b[0m_fillcolor\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fillcolor\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2245\u001b[0;31m         return (self._hatch or\n\u001b[0m\u001b[1;32m   2246\u001b[0m                 (_fillcolor is not None and\n\u001b[1;32m   2247\u001b[0m                  (len(_fillcolor) <= 3 or _fillcolor[3] != 0.0)))\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "hc_samples(input_gene_expression=\"https://datasets.genepattern.org/data/TCGA_BRCA_temp/TCGA_BRCA_all_matched.filtered.gct\", clustering_type=\"Single\", distance_metric=\"pearson\", file_basename=\"HC_out_2\", clusters_to_highlight=6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "78053bfce21b450f9fb7165d248fa743",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00057')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nmfconsensus_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00057')\n",
    "nmfconsensus_job_spec = nmfconsensus_task.make_job_spec()\n",
    "nmfconsensus_job_spec.set_parameter(\"dataset.filename\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18035/BRCA.gct\")\n",
    "nmfconsensus_job_spec.set_parameter(\"k.initial\", \"2\")\n",
    "nmfconsensus_job_spec.set_parameter(\"k.final\", \"6\")\n",
    "nmfconsensus_job_spec.set_parameter(\"num.clusterings\", \"20\")\n",
    "nmfconsensus_job_spec.set_parameter(\"max.num.iterations\", \"2000\")\n",
    "nmfconsensus_job_spec.set_parameter(\"error.function\", \"divergence\")\n",
    "nmfconsensus_job_spec.set_parameter(\"random.seed\", \"123456789\")\n",
    "nmfconsensus_job_spec.set_parameter(\"output.file.prefix\", \"<dataset.filename_basename>\")\n",
    "nmfconsensus_job_spec.set_parameter(\"stop.convergence\", \"40\")\n",
    "nmfconsensus_job_spec.set_parameter(\"stop.frequency\", \"10\")\n",
    "genepattern.GPTaskWidget(nmfconsensus_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f55d72cfb09d4b94859ea66ebc9e8508",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18037)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18037 = gp.GPJob(genepattern.get_session(0), 18037)\n",
    "genepattern.GPJobWidget(job18037)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "genepattern": {
     "type": "task"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0ff7d38ef9b3454fb7f0b7a93e0c9bf2",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPTaskWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPTaskWidget(lsid='urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00057')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nmfconsensus_task = gp.GPTask(genepattern.get_session(0), 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00057')\n",
    "nmfconsensus_job_spec = nmfconsensus_task.make_job_spec()\n",
    "nmfconsensus_job_spec.set_parameter(\"dataset.filename\", \"https://gp-beta-ami.genepattern.org/gp/jobResults/18036/BRCA.preprocessed.gct\")\n",
    "nmfconsensus_job_spec.set_parameter(\"k.initial\", \"2\")\n",
    "nmfconsensus_job_spec.set_parameter(\"k.final\", \"6\")\n",
    "nmfconsensus_job_spec.set_parameter(\"num.clusterings\", \"20\")\n",
    "nmfconsensus_job_spec.set_parameter(\"max.num.iterations\", \"2000\")\n",
    "nmfconsensus_job_spec.set_parameter(\"error.function\", \"divergence\")\n",
    "nmfconsensus_job_spec.set_parameter(\"random.seed\", \"123456789\")\n",
    "nmfconsensus_job_spec.set_parameter(\"output.file.prefix\", \"<dataset.filename_basename>\")\n",
    "nmfconsensus_job_spec.set_parameter(\"stop.convergence\", \"40\")\n",
    "nmfconsensus_job_spec.set_parameter(\"stop.frequency\", \"10\")\n",
    "genepattern.GPTaskWidget(nmfconsensus_task)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "genepattern": {
     "type": "job"
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4f35c5abe24e4712bef855ba8704618f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>GPJobWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in the Jupyter Notebook or JupyterLab Notebook, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "GPJobWidget(job_number=18041)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "job18041 = gp.GPJob(genepattern.get_session(0), 18041)\n",
    "genepattern.GPJobWidget(job18041)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.6",
   "language": "python",
   "name": "python3.6"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
